frontend/mainwindow.cpp

```diff
@@ -229,12 +229,18 @@ std::vector<std::pair<QString, QString>> MainWindow::getFields(QString instrumen
 {
     std::vector<std::pair<QString, QString>> desiredInstFields;
     QDomNodeList desiredInstrumentFields;
-
-    QFile file(":/tableConfig.xml");
-    file.open(QIODevice::ReadOnly);
     QDomDocument dom;
-    dom.setContent(&file);
-    file.close();
+
+    QSettings settings(QSettings::IniFormat, QSettings::UserScope, "ISIS", "jv2");
+    if (settings.value("tableConfig", "fail").toString() == "fail")
+    {
+        QFile file(":/tableConfig.xml");
+        file.open(QIODevice::ReadOnly);
+        dom.setContent(&file);
+        file.close();
+    }
+    else
+        dom.setContent(settings.value("tableConfig", "fail").toString());
```

Comment:
Looks like this code is repeated three times and would be better off in a separate function.